contracts/voucher/Voucher.sol

```diff
@@ -35,8 +32,14 @@ contract Voucher is IVoucher, Upgradeable, ERC721Upgradeable, Utils {
     // a mapping between an owner to its tokenIds
     mapping(address => EnumerableSet.UintSet) internal _ownedTokens;
 
+    // controller address - used to mint / burn
+    address private _controller;
+
+    // a flag indicating whether the controller role has been set
+    bool private _controllerSet;
```

Comment:
Why not just checking that `_controller` isn't `0x0`?

Reply:
changed